src/core/public/apm_system.ts

```diff
@@ -73,4 +78,52 @@ export class ApmSystem {
       }
     });
   }
+
+  /**
+   * Adds a filter to the APM configuration for normalizing transactions of the 'http-request' type to remove the
+   * hostname, protocol, port, and base path. Allows for coorelating data cross different deployments.
+   */
+  private addHttpRequestNormalization(apm: ApmBase) {
+    apm.addFilter((payload) => {
+      payload.transactions
+        .filter((t: { type: string }) => t.type === 'http-request')
+        .forEach((t: { name: string }) => {
+          /** Split URLs of the from "GET protocol://hostname:port/pathname" into method & hostname */
+          const matches = t.name.match(HTTP_REQUEST_TRANSACTION_NAME_REGEX);
+          if (!matches) {
+            return;
+          }
+
+          const [, method, originalUrl] = matches;
+          // Normalize the URL
+          const normalizedUrl = modifyUrl(originalUrl, (parts) => {
```

Comment:
Do we need this? RUM agent already strips out everything in the URL except for the path when the URL is relative and only keeps the full URL for absolute url's. 

https://github.com/elastic/apm-agent-rum-js/blob/4de1490c6a08260423d42176d999819408d158e5/packages/rum-core/src/performance-monitoring/performance-monitoring.js#L261-L263

May be we can fix it there if we see bugs? I am also fine with your approach if we need fine tuned behaviours.

Reply:
It seems like there may be a bug here. The `requestUrl` that RUM receives includes the host even though the `Request` object that was passed to the `fetch` API was not configured with an absolute URL. It seems that when using `new Request` with a relative URL, the browser injects the current origin into the URL on the resulting object. Here are screenshots from DevTools on these three locations for the same request:

### URL passed to new Request()

<img width="296" alt="image" src="https://user-images.githubusercontent.com/1813008/95795031-9b078d80-0ca6-11eb-88d4-023401242b54.png">

### Request.url

<img width="453" alt="image" src="https://user-images.githubusercontent.com/1813008/95795064-b1154e00-0ca6-11eb-9d6d-2ed5aa89990b.png">

### requestUrl object in rum-core

<img width="563" alt="image" src="https://user-images.githubusercontent.com/1813008/95795088-bf636a00-0ca6-11eb-9869-b4e064480562.png">

---

Regardless of this bug, we will still need to strip the basePath from the pathname. I think I will go ahead and update this code to be able to handle relative URLs as well so this does not break if/when this RUM bug is fixed.

Reply:
ahh thanks for the info, wont consider it as bug, but definitely we can handle this in the RUM agent. Sure that makes sense.

Reply:
Yes you are right, if string is passed, our custom URL parser parses and keep them as `relative`, But since the Request spec as you mentioned above passes the complete URL, we don't handle it correctly at the moment. I don't know whether to call it as bug as the URL is instead the complete URL when Request is passed to `fetch`. But we can definitely fix it in the agent by checking for origin URL.